### PR TITLE
Fix memory leak.

### PR DESCRIPTION
**Pull request by llapx**
Signed-off-by: Lei Liu <Lei.Liu.AP@sony.com>

### CODE REVIEW

**Review by aprotyas — APPROVED** [review]
LGTM under #822's context.

Although, I wonder if there was a case to capture this concern through a unit test as well? I'm not sure, but this seems like a case that should be tested for (i.e. is the client/server correctly "taking" the response/request).

**Comment by llapx** [comment]
In my option, it's a bug which may affect many test cases.

**Comment by aprotyas** [comment]
For reviewers, just reproducing the memory usage charts from https://github.com/ros2/rclpy/issues/822#issuecomment-923655611
test_client_before
test_client_after

The same observations should hold for services using `rclpy`.
As such, I think this closes #822.

File: rclpy/src/rclpy/client.cpp [hunk 1]
```diff
@@ -138,7 +138,6 @@ Client::take_response(py::object pyresponse_type)
 
   result_tuple[1] = convert_to_py(taken_response.get(), pyresponse_type);
   // result_tuple now owns the message
```
Comment (iuhilnehc-ynos) [inline rclpy/src/rclpy/client.cpp:140]: There is no reason to keep this comment, I suggest deleting it.

**Review by iuhilnehc-ynos — COMMENTED** [review]
nitpick.

**Review by fujitatomoya — APPROVED** [review]
lgtm with @iuhilnehc-ynos 's comment.

**Comment by fujitatomoya** [comment]
@aprotyas it seems that memory usage doe not change on the graph chart between before and after? it still increases the memory usage constantly on client side? just checking.

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15324/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10035/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13004/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15517/)

**Comment by aprotyas** [comment]
> @aprotyas it seems that memory usage doe not change on the graph chart between before and after? it still increases the memory usage constantly on client side? just checking.

No no, the first image is from before, and the second image is from after this PR.
In the second image, you see that the memory usage does not continuously increase (stays consistent around 60 MB). So I believe the PR fixes the memory leak.

**Comment by fujitatomoya** [comment]
@aprotyas Ah, i see it. thanks!

**Review by iuhilnehc-ynos — APPROVED** [review]
LGTM.

Re-CI (error happened on Window because of rti package installation, rebuild with @fujitatomoya setting on ci_launcher):
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15333/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10044/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13013/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15527/)

**Pull request merged**